packages/xpath/src/lib/collections/sort.ts

```diff
@@ -15,11 +15,18 @@ class UnseededPseudoRandomNumberGenerator implements PseudoRandomNumberGenerator
 }
 
 class SeededPseudoRandomNumberGenerator implements PseudoRandomNumberGenerator {
+	// Park-Miller PRNG
 	protected seed: number;
 
-	constructor(seed: Int) {
-		let initialSeed = seed % SEED_MODULO_OPERAND;
-
+	constructor(seed: Int | bigint) {
+		let initialSeed: number;
+		if (typeof seed === 'bigint') {
+			// the result of the modulo operation is always smaller than Number.MAX_SAFE_INTEGER,
+			// thus it's safe to convert to a Number.
+			initialSeed = Number(BigInt(seed) % BigInt(SEED_MODULO_OPERAND));
+		} else {
+			initialSeed = Number(seed) % Number(SEED_MODULO_OPERAND);
```

Comment:
```suggestion
			initialSeed = seed % SEED_MODULO_OPERAND;
```

Both operands are already `number` here.